src/spikeinterface/qualitymetrics/misc_metrics.py

```diff
@@ -389,7 +389,10 @@ def compute_refrac_period_violations(
     nb_violations = {}
     rp_contamination = {}
 
-    for i, unit_id in enumerate(unit_ids):
+    for i, unit_id in enumerate(sorting.unit_ids):
```

Comment:
What about computing nb_rp_violations only on `unit_ids` and not on `sorting.unit_ids` and have a
```
if unit_ids is None:
    unit_ids = sorting.unit_ids
```
?

Reply:
I thought about it, but then this requires doing `sorting.select_units(unit_ids).to_spike_vector()`

And this means recomputing the spike vector, which I worry might be slower.
We could maybe do `sorting.to_spike_vector()` and then remove any indices not in unit_ids?

Reply:
Makes sense. The most elegant option would be to override the `UnitsSelectionSorting` so that if the parent is cached, it slices it and then go for the `sorting.select_units(unit_ids).to_spike_vector()`.

I'm also fine with this "simpler" implementation. Let me know what you prefer!

Reply:
I think overriding would be the best option, but every time I mention overriding a method, @samuelgarcia doesn't like it ^^'

Reply:
Bad idea.
And also this will not as easy because you will need to move all the unit_index to the new_index!